src/VisualStudio/Core/Def/Implementation/SyncNamespaces/SyncNamespacesCommandHandler.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Design;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SyncNamespaces;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SyncNamespaces
+{
+    [Export(typeof(SyncNamespacesCommandHandler)), Shared]
+    internal sealed class SyncNamespacesCommandHandler
+    {
+        private readonly VisualStudioWorkspace _workspace;
+        private readonly IUIThreadOperationExecutor _threadOperationExecutor;
+        private IServiceProvider? _serviceProvider;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public SyncNamespacesCommandHandler(
+            IUIThreadOperationExecutor threadOperationExecutor,
+            VisualStudioWorkspace workspace)
+        {
+            _threadOperationExecutor = threadOperationExecutor;
+            _workspace = workspace;
+        }
+
+        public void Initialize(IServiceProvider serviceProvider)
+        {
+            Contract.ThrowIfNull(serviceProvider);
+
+            _serviceProvider = serviceProvider;
+
+            // Hook up the "Remove Unused References" menu command for CPS based managed projects.
+            var menuCommandService = (IMenuCommandService)_serviceProvider.GetService(typeof(IMenuCommandService));
+            if (menuCommandService != null)
+            {
+                VisualStudioCommandHandlerHelpers.AddCommand(menuCommandService, ID.RoslynCommands.SyncNamespaces, Guids.RoslynGroupId, OnSyncNamespacesForSelectedProject, OnSyncNamespacesForSelectedProjectStatus);
+            }
+        }
+
+        private void OnSyncNamespacesForSelectedProjectStatus(object sender, EventArgs e)
+        {
+            var command = (OleMenuCommand)sender;
+
+            var visible = false;
+            var enabled = false;
+
+            if (VisualStudioCommandHandlerHelpers.TryGetSelectedProjectHierarchy(_serviceProvider, out var projectHierarchy))
```

Comment:
- What happens when I select more than 1 project? 
- What happens when I select a solution folder?

Reply:
> - What happens when I select more than 1 project?

That is a good question because it would be nice to update both.

> - What happens when I select a solution folder?

It  will  update all the C# projects.

Reply:
Sorry, misread solution folder. Currently it won't be offered.

Reply:
We currently do not display this command when there is a multi-selection, but I could see that being a future enhancement.